libraries/chain/controller.cpp

```diff
@@ -2644,7 +2644,8 @@ struct controller_impl {
 
       transaction_checktime_timer trx_timer(timer);
       const packed_transaction trx( std::move( etrx ) );
-      transaction_context trx_context( self, trx, trx.id(), std::move(trx_timer), bb.action_receipt_digests().store_which(), start );
+      transaction_context trx_context( self, trx, trx.id(), std::move(trx_timer), bb.action_receipt_digests().store_which(),
+                                       start, transaction_metadata::trx_type::implicit );
```

Comment:
onblock is implicit. But I don't see it see anywhere.

Reply:
https://github.com/AntelopeIO/spring/blob/f0d379a83309387d4e64e000823c928e20257396/libraries/chain/controller.cpp#L3031-L3031

It is set via `trx->get_trx_type()`.